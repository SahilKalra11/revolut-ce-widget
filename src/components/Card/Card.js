import React, { Component } from "react";
import "./Card.scss";
import Currency from "../Currency";
import { Select, MenuItem } from "@material-ui/core";
import ExchangeInput from "../ExchangeInput";
import { getCurrencySymbol } from "../../helpers";

class Card extends Component {
  render() {
    const {
      currency,
      handleCurrencyChange,
      cardId,
      balance,
      value,
      handleValueChange,
      currencyFrom,
      rates,
      autoFocus,
    } = this.props;
    return (
      <section className="card-slider">
        <div className="exchange">
          <Select
            labelId="label"
            id="select"
            value={currency}
            onChange={handleCurrencyChange}
          >
            {this.props.currencyList.map((currencyItem) => {
              const key = `${currencyItem}${cardId}`;
              return (
                <MenuItem value={currencyItem} key={key}>
                  <Currency currency={currencyItem} />
                </MenuItem>
              );
            })}
          </Select>
          <div className="exchange__currency-wallet">
            You have {getCurrencySymbol(currency)}
            {balance[currency]}
          </div>
        </div>
        <ExchangeInput
          focus={autoFocus}
          value={value}
          onChange={handleValueChange}
          cardId={cardId}
          currency={currency}
          currencyFrom={currencyFrom}
          rates={rates}
        />
      </section>
    );
  }
}

export default Card;
