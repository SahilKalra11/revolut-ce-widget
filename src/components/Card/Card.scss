@import '../../utilities/variables';
$border: 3vh;

.card-slider {
  
  text-align: center;
  position: relative;
  height: 50%;
  transform-origin: center center;
  display: flex;
  align-items: center;
  padding: 0 $padding;
  box-sizing: border-box;
  color: $color-white !important;
  &:last-child{
    background: rgba(0, 0, 0, 0.2);
  }
  &:first-child::after {
    width: 0;
    height: 0;
    border-left: $border solid transparent;
    border-right: $border solid transparent;
    margin: 0 auto;
    border-top: $border solid $color-blue;
    position: absolute;
    bottom: -$border;
    content: '';
    z-index: 1;
    left: 0;
    right: 0;
  }
}

.exchange {
  flex: 50%;
  font-size: $font-s;
  text-align: left;
}

.exchange__currency-wallet {
  margin-top: $margin;
}

.MuiInputBase-root.MuiInput-root {
  line-height: $font-xl;
  color: $color-white;
}

.MuiSvgIcon-root {
  fill: $color-white !important;
}

.MuiList-root.MuiList-padding {
  padding: 0;
}


.MuiMenuItem-root .exchange__currency-name {
  color: $color-light-blue !important;
  &:hover {
    color: $color-blue !important;
  }
}
