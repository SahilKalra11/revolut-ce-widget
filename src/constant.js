export const AVAILABLE_CURRENCY_LIST = ['USD', 'GBP', 'EUR'];
export const APP_ID = '115aba0dc4cb4ceaa4c38c5104e3cce7';
export const BASE_URL = 'https://openexchangerates.org/api/latest.json?app_id=';
export const TEST_RATES = {USD: 1, EUR: 1.20, GBP: 1.3};
export const TEST_WALLET = {USD: 100, EUR: 100, GBP: 100,};

