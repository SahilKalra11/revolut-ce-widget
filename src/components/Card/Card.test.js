import React from 'react';
import ReactDOM from 'react-dom';
import render from 'react-test-renderer';
import Card from './Card';
import {unmountComponentAtNode} from "react-dom";
import {AVAILABLE_CURRENCY_LIST} from "../../constant";

describe('Card', () => {
    let container = null;
    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    const props = {
        currencyList: AVAILABLE_CURRENCY_LIST,
        cardId: "cardId",
        balance: {
            USD: 100,
            EUR: 100,
            GBP: 100,
        },
        currency: 'USD',
        currencyFrom: "USD",
        value: 10,
        rates: {
            USD: 1,
            EUR: 1.20,
            GBP: 1.3
        },
        handleCurrencyChange: jest.fn(),
        handleValueChange: jest.fn(),
    };

    it('renders component without crashing', () => {
        ReactDOM.render(
            <Card {...props} />,
            container
        );
    });

    test('matches to snapshot', () => {
        const component = render.create(
            <Card {...props} />
        );
        let componentTree = component.toJSON();
        expect(componentTree).toMatchSnapshot();
    });
});
