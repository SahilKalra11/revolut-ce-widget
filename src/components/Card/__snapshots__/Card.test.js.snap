// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches to snapshot 1`] = `
<section
  className="card-slider"
>
  <div
    className="exchange"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="label select"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="select"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <section
          className="exchange__currency"
        >
          <div
            className="exchange__currency-name"
          >
            USD
          </div>
        </section>
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        tabIndex={-1}
        value="USD"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <div
      className="exchange__currency-wallet"
    >
      You have 
      $
      100
    </div>
  </div>
  <section
    className="exchange__action"
  >
    <input
      className="exchange__input"
      onChange={[MockFunction]}
      type="text"
      value={10}
    />
    <div
      className="exchange__course"
    >
      
    </div>
  </section>
</section>
`;
